docs/api/hooks.md

```diff
@@ -67,19 +69,37 @@ export const CounterComponent = () => {
 }
 ```
 
-Using props to determine what to extract:
+Using props via closure to determine what to extract:
 
 ```jsx
 import React from 'react'
 import { useSelector } from 'react-redux'
 
 export const TodoListItem = props => {
-  const todo = useSelector(state => state.todos[props.id], [props.id])
-
+  const todo = useSelector(state => state.todos[props.id])
   return <div>{todo.text}</div>
 }
 ```
 
+Using `deps` to ensure the same selector is used in each render:
+
+```jsx
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { createSelector } from 'reselect'
+
+const allOtherTodos = createSelector(
+  state => state.todos,
+  (_, id) => id,
+  (todos, id) => todos.filter(todo => todo.id !== id)
+)
+
+export const OtherTodoListItems = ({ id }) => {
+  const otherTodos = useSelector(s => allOtherTodos(s, id), [id])
```

Comment:
I don't have an in-depth understating of `useSelector` so I might be wrong, but I think that in this example the same `allOtherTodos` selector will be used in each render irrespectively if `deps` are provided or not.  This is because the selector is created outside of `OtherTodoListItems`, so it's reference never changes. The arrow function wrapping it will/will not change, but it does not really matter.

Additionally, in this example the selector will be shared across all instances of this `OtherTodoListItems`. And because reselect selectors have a cache of 1, this mean it will always recompute (see last paragraph from https://github.com/reduxjs/reselect#accessing-react-props-in-selectors )

Reply:
You are of course right. When writing this example I was thinking whether I should use a factory selector instead but thought that might be too much added complexity. However, it seems that is necessary for the selector to work correctly. I'll adjust the example.

Reply:
I decided to just move the selector into the component after all to keep it simple

Reply:
I had more thoughts on the subject of `deps`. I have written them down in https://github.com/reduxjs/react-redux/issues/1252#issuecomment-488591342